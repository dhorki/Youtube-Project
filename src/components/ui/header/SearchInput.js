import React from 'react';
import styled from 'styled-components';
import { BsSearch } from 'react-icons/bs';
import { useForm } from '../../../hooks/useForm';
import { MdClear } from 'react-icons/md';
import { useTheme } from '../../../hooks/useTheme';

const StyledSearchBox = styled.form`
  background-color: ${({ theme }) => theme.inputBackground};
  border-radius: 2px;
  border: 1px solid ${({ theme }) => theme.inputBorder};
  display: flex;
  flex-direction: row;
  margin: auto;
  height: 30px;

  > input {
    color: ${({ theme }) => theme.primary};
    font-family: 'Roboto', 'Arial', sans-serif;
    border: none;
    border-radius: 2px;
    background-color: transparent;
    padding: 0.5rem;
    padding-right: 0.2rem;
    width: ${({ q }) => (q.length === 0 ? 380 : 350)}px;

    &:focus {
      outline: none;
    }
  }

  > .clear-button {
    ${({ q }) => q.length === 0 && 'display: none;'}
    width: 30px;
    color: ${({ theme }) => theme.primary};
    background-color: transparent;
    border: none;
    cursor: pointer;
    height: 100%;
    margin-top: auto;
    margin-bottom: auto;
  }

  > .search-button {
    color: ${({ theme }) => theme.primary};
    background-color: ${({ theme }) => theme.searchButtonBackground};
    border: none;
    border-left: 1px solid ${({ theme }) => theme.inputBorder};
    cursor: pointer;
    justify-content: center;
    padding: 0;
    margin: auto;
    width: 50px;
    height: 100%;
  }
`;

export const SearchInput = () => {
  const { getTheme } = useTheme();
  const theme = getTheme();
  // TODO: initialize the form with the initial values
  const [{ q }, handleInputChange] = useForm({ q: '' });

  const handleSearch = (e) => {
    e.preventDefault();
    console.log('Search', q);
  };

  const handleClear = (e) => {
    e.preventDefault();
    handleInputChange({ target: { name: 'q', value: '' } });
  };

  return (
    <StyledSearchBox theme={theme} q={q}>
      <input
        type="text"
        autoComplete="off"
        placeholder="Search"
        value={q}
        onChange={handleInputChange}
        name="q"
        theme={theme}
      />
      <button className="clear-button" type="button" onClick={handleClear}>
        <MdClear />
      </button>

      <button className="search-button" type="submit" onClick={handleSearch}>
        <BsSearch />
      </button>
    </StyledSearchBox>
  );
};
