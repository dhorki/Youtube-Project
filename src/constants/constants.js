export const styles = {
  colors: {
    light: {
      appIcon: '#ff0200',
      background: '#F9F9F9',
      barBackground: '#ffffff',
      primary: '#030303',
      secondary: '#606060',
      linkActive: '#ff0200',
      linkHoverBackground: '#ececec',
      inputBackground: '#ffffff',
      inputBorder: '#cccccc',
      searchButtonBackground: '#f8f8f8',
      // google button
      white: '#fff',
      googleBlue: '#4285f4',
      buttonActiveBlue: '#1669f2',
    },
    dark: {
      appIcon: '#ff0200',
      background: '#181818',
      barBackground: '#212121',
      primary: '#ffffff',
      secondary: '#aaaaaa',
      linkActive: '#ff0200',
      linkHoverBackground: '#303030',
      inputBackground: '#121212',
      inputBorder: '#303030',
      searchButtonBackground: '#323232',
      // google button
      white: '#fff',
      googleBlue: '#4285f4',
      buttonActiveBlue: '#1669f2',
    },
  },
  props: {
    transitionTime: '0.4s',
    sidebarWidth: '150px',
  },
};

export const screenSizes = {
  xsmall: '675px',
  small: '830px',
  mid: '1200px',
};

export const mainViewTypes = {
  searchGallery: 'searchGallery',
  favoritesGallery: 'favoritesGallery',
  videoDetails: 'videoDetails',
};

export const environmentActions = {
  setTheme: 'setTheme',
  setSidebarShow: 'setSidebarShow',
  setSearchQuery: 'setSearchQuery',
  setModalShow: 'setModalShow',
  setUser: 'setUser',
};

export const modalTypes = {
  login: 'login',
  register: 'register',
  resetPassword: 'reset-password',
  none: 'none',
};

export const resetSearchQueryAction = {
  type: environmentActions.setSearchQuery,
  payload: {
    q: '',
  },
};

export const hideModalAction = {
  type: environmentActions.setModalShow,
  payload: {
    modalShow: modalTypes.none,
  },
};
