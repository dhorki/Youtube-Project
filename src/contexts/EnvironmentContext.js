import { createContext } from 'react';

export const EnvironmentContext = createContext();
