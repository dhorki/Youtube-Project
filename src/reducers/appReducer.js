import { environmentActions } from '../constants/constants';

export const appReducer = (state, { type, payload }) => {
  switch (type) {
    case environmentActions.setTheme:
      return { ...state, theme: payload.theme };
    case environmentActions.setSidebarShow:
      return { ...state, sidebarShow: payload.sidebarShow };
    case environmentActions.setSearchQuery:
      return { ...state, q: payload.q };
    default:
      return state;
  }
};
