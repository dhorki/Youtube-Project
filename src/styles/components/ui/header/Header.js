import styled from 'styled-components';
import { styles } from '../../../../constants/constants';

export const StyledHeader = styled.header`
  background-color: ${({ theme }) => theme.barBackground};
  box-shadow: 0 2px 4px 0 ${({ theme }) => theme.inputBorder};
  display: flex;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;

  transition: background-color ${styles.props.transitionTime} ease;

  > button {
    color: ${({ theme }) => theme.primary};
    font-size: 15px;
    font-weight: normal;
    background-color: transparent;
    border: none;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 0;
    width: 2rem;
  }

  > .app-link {
    text-decoration: none;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-top: auto;
    margin-bottom: auto;

    > .app-icon {
      font-size: 20px;
      color: ${({ theme }) => theme.appIcon};
      display: flex;
      flex-direction: row;
      justify-content: center;
      padding: 0;
      margin: auto;
    }

    > h1 {
      color: ${({ theme }) => theme.primary};
      font-size: 18px;
      font-weight: bolder;
      margin: 0 5px;
    }
  }
`;
