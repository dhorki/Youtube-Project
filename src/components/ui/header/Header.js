import React, { useContext } from 'react';
import { SearchInput } from './SearchInput';
import { UserSettings } from './UserSettings';
import { IoLogoYoutube } from 'react-icons/io5';
import { GiHamburgerMenu } from 'react-icons/gi';
import { BsFillArrowLeftSquareFill } from 'react-icons/bs';
import { Link } from 'react-router-dom';
import { EnvironmentContext } from '../../../contexts/EnvironmentContext';
import { StyledHeader } from '../../../styles/components/ui/header/Header';
import { environmentActions } from '../../../constants/constants';
import { resetSearch } from '../../../helpers/resetSearch';

export const Header = () => {
  const { environment, dispatchEnv } = useContext(EnvironmentContext);
  const { theme, sidebarShow } = environment;

  const sidebarShowHandler = () => {
    const action = {
      type: environmentActions.setSidebarShow,
      payload: {
        sidebarShow: !sidebarShow,
      },
    };

    dispatchEnv(action);
  };

  return (
    <StyledHeader theme={theme}>
      <button id="sidebarToggler" onClick={sidebarShowHandler}>
        {sidebarShow ? <BsFillArrowLeftSquareFill /> : <GiHamburgerMenu />}
      </button>
      <Link className="app-link" to="/" onClick={() => resetSearch(dispatchEnv)}>
        <div className="app-icon">
          <IoLogoYoutube />
        </div>
        <h1>Llutuve</h1>
      </Link>
      <SearchInput />
      <UserSettings />
    </StyledHeader>
  );
};
