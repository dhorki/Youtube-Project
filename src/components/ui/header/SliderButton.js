import React, { useContext, useState } from 'react';
import PropTypes from 'prop-types';
import { StyledSliderButton } from '../../../styles/components/ui/header/SliderButton';
import { EnvironmentContext } from '../../../contexts/EnvironmentContext';

export const SliderButton = ({ label, initState, callback }) => {
  const { environment } = useContext(EnvironmentContext);
  const { theme } = environment;

  const [hover, setHover] = useState(false);
  const [checked, setChecked] = useState(initState);

  const handleToggle = () => {
    setChecked(!checked);
    if (callback) {
      callback(!checked);
    }
  };

  return (
    <StyledSliderButton
      onMouseOver={() => {
        setHover(true);
      }}
      onMouseOut={() => {
        setHover(false);
      }}
      onClick={handleToggle}
      hover={hover}
      checked={checked}
      theme={theme}
    >
      <div className="slider-background">
        <div className="slider" />
      </div>
      {label && <div className="slider-label">{label}</div>}
    </StyledSliderButton>
  );
};

SliderButton.propTypes = {
  label: PropTypes.string,
  initState: PropTypes.bool.isRequired,
  callback: PropTypes.func,
};
