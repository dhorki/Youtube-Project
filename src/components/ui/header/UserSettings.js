import React, { useContext, useState } from 'react';
import { RiLoginBoxFill, RiLogoutBoxFill } from 'react-icons/ri';
import { SliderButton } from './SliderButton';
import { EnvironmentContext } from '../../../contexts/EnvironmentContext';
import { environmentActions, modalTypes, styles } from '../../../constants/constants';
import { StyledUserSettings } from '../../../styles/components/ui/header/UserSettings';
import { logout } from '../../../firebase';

export const UserSettings = () => {
  const { environment, dispatchEnv } = useContext(EnvironmentContext);
  const { theme, user } = environment;

  const [submenuShow, setsubmenuShow] = useState(false);

  const handleThemeChange = (checked) => {
    const action = {
      type: environmentActions.setTheme,
      payload: {
        theme: checked ? styles.colors.dark : styles.colors.light,
      },
    };

    dispatchEnv(action);
  };

  const handleLoginLogout = (toLogin) => {
    const action = {
      type: environmentActions.setModalShow,
      payload: {
        // if not toLogin, then logout, logout does not need modalShow
        modalShow: toLogin ? modalTypes.login : modalTypes.none,
      },
    };

    if (toLogin) {
      console.log('login');
    } else {
      const action = {
        type: environmentActions.setUser,
        payload: {
          user: null,
        },
      };

      logout();
      dispatchEnv(action);
    }

    setsubmenuShow(false);

    dispatchEnv(action);
  };

  const logged = false;

  return (
    <StyledUserSettings theme={theme} submenuShow={submenuShow}>
      <SliderButton
        label="Dark Mode"
        initState={theme === styles.colors.dark}
        callback={handleThemeChange}
      />
      {user && <div className="username-tag">{user.name}</div>}
      <img
        className="user-badge"
        onClick={() => setsubmenuShow(!submenuShow)}
        alt={user?.name}
        src={
          user?.photo
            ? user.photo
            : 'https://cdn.pixabay.com/photo/2016/11/08/15/21/user-1808597_1280.png'
        }
      />
      <div className="submenu-container">
        <div className="submenu-item" onClick={() => handleLoginLogout(null === user)}>
          <div className="submenu-icon">
            {logged ? <RiLogoutBoxFill /> : <RiLoginBoxFill />}
          </div>
          <div className="submenu-label">{null === user ? 'Login' : 'Logout'}</div>
        </div>
      </div>
    </StyledUserSettings>
  );
};
