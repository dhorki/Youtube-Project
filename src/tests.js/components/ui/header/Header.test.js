import React, { useReducer } from 'react';
import { Router } from 'react-router-dom';
import { renderHook, act } from '@testing-library/react-hooks';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import { styles } from '../../../../constants/constants';
import { EnvironmentContext } from '../../../../contexts/EnvironmentContext';
import { Header } from '../../../../components/ui/header/Header';
import { appReducer } from '../../../../reducers/appReducer';

describe('Header', () => {
  const environment = {
    theme: styles.colors.light,
    sidebarShow: false,
    q: '',
  };

  it('should render correctly with collapsed sidebar', () => {
    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <Router history={createBrowserHistory()}>
          <Header />
        </Router>
      </EnvironmentContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with expanded sidebar', () => {
    environment.sidebarShow = true;

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment }}>
        <Router history={createBrowserHistory()}>
          <Header />
        </Router>
      </EnvironmentContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch set sidebar show action on button click', () => {
    const { result } = renderHook(() => useReducer(appReducer, environment));
    const [prevEnv, dispatchEnv] = result.current;
    const { sidebarShow: prevSidebarShow } = prevEnv;

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment: prevEnv, dispatchEnv }}>
        <Router history={createBrowserHistory()}>
          <Header />
        </Router>
      </EnvironmentContext.Provider>
    );

    act(() => {
      wrapper.find('#sidebarToggler').simulate('click');
    });

    const [env] = result.current;
    const { sidebarShow } = env;

    expect(sidebarShow).toEqual(!prevSidebarShow);
  });

  it('should reset search query on app link click', () => {
    environment.q = 'some search query';

    const { result } = renderHook(() => useReducer(appReducer, environment));
    const [prevEnv, dispatchEnv] = result.current;
    const { q: prevQ } = prevEnv;

    const wrapper = mount(
      <EnvironmentContext.Provider value={{ environment: prevEnv, dispatchEnv }}>
        <Router history={createBrowserHistory()}>
          <Header />
        </Router>
      </EnvironmentContext.Provider>
    );

    act(() => {
      wrapper.find('a.app-link').simulate('click');
    });

    const [env] = result.current;
    const { q } = env;

    expect(q).not.toEqual(!prevQ);
    expect(q.length).toEqual(0);
  });
});
