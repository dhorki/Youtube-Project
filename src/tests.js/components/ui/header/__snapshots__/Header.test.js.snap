// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly with collapsed sidebar 1`] = `
<header
  className="sc-hKMtZM hjNxSb"
>
  <button
    id="sidebarToggler"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        key="0"
      />
    </svg>
  </button>
  <a
    className="app-link"
    href="/"
    onClick={[Function]}
  >
    <div
      className="app-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M508.64 148.79c0-45-33.1-81.2-74-81.2C379.24 65 322.74 64 265 64h-18c-57.6 0-114.2 1-169.6 3.6C36.6 67.6 3.5 104 3.5 149 1 184.59-.06 220.19 0 255.79q-.15 53.4 3.4 106.9c0 45 33.1 81.5 73.9 81.5 58.2 2.7 117.9 3.9 178.6 3.8q91.2.3 178.6-3.8c40.9 0 74-36.5 74-81.5 2.4-35.7 3.5-71.3 3.4-107q.34-53.4-3.26-106.9zM207 353.89v-196.5l145 98.2z"
          key="0"
        />
      </svg>
    </div>
    <h1>
      Llutuve
    </h1>
  </a>
  <form
    className="sc-bczRLJ byqFuz"
  >
    <input
      autoComplete="off"
      name="q"
      onChange={[Function]}
      placeholder="Search"
      theme={
        Object {
          "appIcon": "#ff0200",
          "background": "#F9F9F9",
          "barBackground": "#ffffff",
          "inputBackground": "#ffffff",
          "inputBorder": "#cccccc",
          "linkActive": "#ff0200",
          "linkHoverBackground": "#ececec",
          "primary": "#030303",
          "searchButtonBackground": "#f8f8f8",
          "secondary": "#606060",
        }
      }
      type="text"
      value=""
    />
    <button
      className="clear-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
          key="0"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          key="1"
        />
      </svg>
    </button>
    <button
      className="search-button"
      onClick={[Function]}
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          key="0"
        />
      </svg>
    </button>
  </form>
  <div
    className="sc-dkzDqf cbTxFK"
  >
    <button
      checked={false}
      className="sc-gsnTZi iizJZa"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="slider-background"
      >
        <div
          className="slider"
        />
      </div>
      <div
        className="slider-label"
      >
        Dark Mode
      </div>
    </button>
    <div
      className="username-tag"
    >
      Victor Infante
    </div>
    <img
      alt="Victor Infante"
      className="user-badge"
      src="https://cdn.pixabay.com/photo/2016/11/08/15/21/user-1808597_1280.png"
    />
  </div>
</header>
`;

exports[`Header should render correctly with expanded sidebar 1`] = `
<header
  className="sc-hKMtZM hjNxSb"
>
  <button
    id="sidebarToggler"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 14a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v12zm-4.5-6.5H5.707l2.147-2.146a.5.5 0 1 0-.708-.708l-3 3a.5.5 0 0 0 0 .708l3 3a.5.5 0 0 0 .708-.708L5.707 8.5H11.5a.5.5 0 0 0 0-1z"
        key="0"
      />
    </svg>
  </button>
  <a
    className="app-link"
    href="/"
    onClick={[Function]}
  >
    <div
      className="app-icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M508.64 148.79c0-45-33.1-81.2-74-81.2C379.24 65 322.74 64 265 64h-18c-57.6 0-114.2 1-169.6 3.6C36.6 67.6 3.5 104 3.5 149 1 184.59-.06 220.19 0 255.79q-.15 53.4 3.4 106.9c0 45 33.1 81.5 73.9 81.5 58.2 2.7 117.9 3.9 178.6 3.8q91.2.3 178.6-3.8c40.9 0 74-36.5 74-81.5 2.4-35.7 3.5-71.3 3.4-107q.34-53.4-3.26-106.9zM207 353.89v-196.5l145 98.2z"
          key="0"
        />
      </svg>
    </div>
    <h1>
      Llutuve
    </h1>
  </a>
  <form
    className="sc-bczRLJ byqFuz"
  >
    <input
      autoComplete="off"
      name="q"
      onChange={[Function]}
      placeholder="Search"
      theme={
        Object {
          "appIcon": "#ff0200",
          "background": "#F9F9F9",
          "barBackground": "#ffffff",
          "inputBackground": "#ffffff",
          "inputBorder": "#cccccc",
          "linkActive": "#ff0200",
          "linkHoverBackground": "#ececec",
          "primary": "#030303",
          "searchButtonBackground": "#f8f8f8",
          "secondary": "#606060",
        }
      }
      type="text"
      value=""
    />
    <button
      className="clear-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
          key="0"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          key="1"
        />
      </svg>
    </button>
    <button
      className="search-button"
      onClick={[Function]}
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          key="0"
        />
      </svg>
    </button>
  </form>
  <div
    className="sc-dkzDqf cbTxFK"
  >
    <button
      checked={false}
      className="sc-gsnTZi iizJZa"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="slider-background"
      >
        <div
          className="slider"
        />
      </div>
      <div
        className="slider-label"
      >
        Dark Mode
      </div>
    </button>
    <div
      className="username-tag"
    >
      Victor Infante
    </div>
    <img
      alt="Victor Infante"
      className="user-badge"
      src="https://cdn.pixabay.com/photo/2016/11/08/15/21/user-1808597_1280.png"
    />
  </div>
</header>
`;
